cpp/ql/src/Security/CWE/CWE-416/UseOfUniquePointerAfterLifetimeEnds.ql

```diff
@@ -30,6 +30,7 @@ where
   outlivesFullExpr(c) and
   not c.isFromUninstantiatedTemplate(_) and
   isUniquePointerDerefFunction(c.getTarget()) and
+  not c.getActualType() instanceof BoolType and
```

Comment:
Is there a specific reason we just exclude bool types, and not just any kind of integer type?

Reply:
Not really, no 🤔 Alternatively we could require that the actual type is a pointer type to be more on the safe side?